metals/src/main/scala/scala/meta/internal/metals/ClassPath.scala

```diff
@@ -0,0 +1,35 @@
+package scala.meta.internal.metals
+import scala.meta.internal.semanticdb.SymbolOccurrence
+
+object ClassPath {
+  final case class TypeSignature(value: String) extends AnyVal
+
+  def toTypeSignature(definition: SymbolOccurrence): TypeSignature = {
+    import scala.meta.internal.semanticdb.Scala._
+    @scala.annotation.tailrec
+    def qualifier(symbol: String, acc: List[String]): String = {
```

Comment:
Did you try using `ownerChain`? It's easy to end in infinite recursion when walking up owner chains

```
      import scala.meta.internal.semanticdb.Scala._
      "a/b/c#".ownerChain.map(_.desc.value)
res0: List[String] = List("_root_", "a", "b", "c")

```

Reply:
Thanks, I didn't know the method!